### PR TITLE
Make Symbol an actual type

### PR DESCRIPTION
**Pull request by jgillick**
I ran into a failing case with `symbol` when it's used to define object keys. Instead of adding additional exception in existing types, I felt like actually defining a type for it will be more flexible in the future.

**Use Case**
```typescript
type MyObject = Record<symbol, string>;

const obj: MyObject = {
  foo: "bar",
};
```

**Expected Schema**
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "MyObject": {
      "additionalProperties": {
        "type": "string"
      },
      "type": "object"
    }
  },
  "$ref": "#/definitions/MyObject"
}
```

The original `symbol` PR (https://github.com/vega/ts-json-schema-generator/pull/598) lead to the following error with that example:
> Error: Unexpected key type "def-alias--315-355--0-356" for type "{ [P in K]: T; }" 
> (expected "UnionType" or "StringType")

### CODE REVIEW

**Comment by domoritz** [comment]
You can run `yarn format` to format the files correctly.

**Comment by jgillick** [comment]
Brilliant. Thank you

File: src/TypeFormatter/ObjectTypeFormatter.ts [hunk 1]
```diff
@@ -81,7 +82,9 @@ export class ObjectTypeFormatter implements SubTypeFormatter {
             type: "object",
             ...(Object.keys(properties).length > 0 ? { properties } : {}),
             ...(required.length > 0 ? { required } : {}),
-            ...(additionalProperties === true || additionalProperties instanceof AnyType
+            ...(additionalProperties === true ||
+            additionalProperties instanceof AnyType ||
+            additionalProperties instanceof SymbolType
```
Comment (domoritz) [inline src/TypeFormatter/ObjectTypeFormatter.ts:87]: To test this, you need to add something to the existing object tests.

File: src/Utils/isAssignableTo.ts [hunk 2]
```diff
@@ -69,6 +70,8 @@ function getPrimitiveType(value: LiteralValue) {
             return new NumberType();
         case "boolean":
             return new BooleanType();
+        case "symbol":
+            return new SymbolType();
```
Comment (domoritz) [inline src/Utils/isAssignableTo.ts:74]: This line wasn't touched. Can you extend a test case or add a test case?

Comment (jgillick) [reply src/Utils/isAssignableTo.ts:74]: Looks like I was being a little overzealous with this addition as symbols literals aren't fully supported in TypeScript yet. It also looks like there would be considerably more work to get it playing right with the parser.

**Pull request merged**